Lib/test/test_posix.py

```diff
@@ -1794,6 +1794,39 @@ def test_dup2(self):
         with open(dupfile) as f:
             self.assertEqual(f.read(), 'hello')
 
+    @unittest.skipUnless(hasattr(os, 'posix_spawnp'), "test needs os.posix_spawnp")
+    def test_posix_spawnp(self):
```

Comment:
I think we should also test posix_spawnp against all the test of posix_spawn. This can be done making the current test class a mixin and changing the function top test in every subclass. More or less this:

* `TestPosixSpawn(unittest.TestCase)` -> `class _PosixSpawnMixin`
* Change all calls to `posix.posix_spawn` to `self.testing_function` or similar.
* Create two new test classes:

```python
@unittest.skipUnless(hasattr(os, 'posix_spawn'), "test needs os.posix_spawn")
class TestPosixSpawn(unittest.TestCase, _PosixSpawnMixin):
    testing_function = posix.posix_spawn


@unittest.skipUnless(hasattr(os, 'posix_spawnp'), "test needs os.posix_spawnp")
class TestPosixSpawnP(unittest.TestCase, _PosixSpawnMixin):
    testing_function = posix.posix_spawnp

...

 support.run_unittest(PosixTester, PosixGroupsTester, TestPosixSpawn, TestPosixSpawnP
``` 

We can add this new test in the `TestPosixSpawnP` class

Reply:
I like the idea.

"Change all calls to posix.posix_spawn to self.testing_function"

bikeshedding: I propose the name "spawn_func" :-)

Reply:
`testing_function = posix.posix_spawn` will fail if the function is not available. You need either to use `getattr(posix, 'posix_spawn', None)` or to add a wrapper method.